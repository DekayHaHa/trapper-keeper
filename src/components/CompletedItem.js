import React from 'react'

export const CompletedItem = (props) => {
	return (
		<div>
			<p className='uncompleted-item'>{props.text}</p><button onClick={() => props.toggle(props.id)}>Check</button>
		</div>
	)
}