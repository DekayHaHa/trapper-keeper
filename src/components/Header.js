import React from 'react';
import CreateNote from '../containers/CreateNote';
import { Link } from 'react-router-dom';

export const Header = () => {
  return (
    <header className="header-container">
      <Link to="/" className='link-style'>
        <h1>Task Mancer</h1>
      </Link>
      {/* <h1>Task Mancer</h1> */}
      <Link to="/api/new-note" className="new-note-btn">
        <CreateNote />
      </Link>
    </header>
  );
};
