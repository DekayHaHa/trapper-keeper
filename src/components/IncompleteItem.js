import React from 'react'

export const IncompleteItem = (props) => {
	return (
		<div>
			<p className='completed-item'>{props.text}</p><button onClick={() => props.toggle(props.id)}>Uncheck</button>
		</div>
	)
}