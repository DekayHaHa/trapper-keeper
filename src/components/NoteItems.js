import React from 'react';
import { Divider, Typography } from '@material-ui/core'
import { withStyles } from '@material-ui/core/styles';

export const NoteItems = ({ noteItems, classes }) => {
    const itemsList = noteItems.map(item => (
      <div iscomplete={JSON.stringify(item.isComplete)} key={item.id} >
        <Typography id={item.id} className={item.isComplete ? classes.strikethrough : ''} style={{ color: '#000' }} >
          {item.text}
        </Typography>
        <Divider light />
      </div>
    ));

  console.log('items list: ', itemsList)
  console.log(classes)
    return (
      <div>
        {itemsList.filter(item => item.props.iscomplete === 'false')}
        <Divider className={classes.divider} />
        {itemsList.filter(item => item.props.iscomplete === 'true')}
      </div>
    );
};

const styles = {
  divider: {
    backgroundColor: '#9445FF'
  },
  strikethrough: {
    textDecoration: 'line-through'
  }
};

export default withStyles(styles)(NoteItems);