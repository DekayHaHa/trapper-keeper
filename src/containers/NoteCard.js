import React from 'react';
import Card from '@material-ui/core/Card';

export const NoteCard = (props) => {
    console.log('props in notecard: ', props);
    // const { classes } = this.props;
  return (
    <div>
    </div>

  )
}
