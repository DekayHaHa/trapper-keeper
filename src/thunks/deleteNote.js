import { hasError, removeNote } from '../actions';

export const deleteNote = (id) => {
  const option = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const url = `http://localhost:3001/api/notes/${id}`;
  return async (dispatch) => {
    try {
      const response = await fetch(url, option);
      if(!response.ok) {
        throw Error(response.statusText);
      }
      dispatch(removeNote(id))
    } catch(error) {
      dispatch(hasError(error.message));
    }
  }
}