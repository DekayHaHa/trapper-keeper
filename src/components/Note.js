import React from 'react';
import { NoteCard } from '../containers/NoteCard';
import { Grid } from '@material-ui/core';

export const Note = (props) => {
  // const renderNoteItems = props.itemsList.map(note => (
  //   <Grid key={Date.now()} item xs={12} sm={6} md={4} lg={2}>
  //     <NoteCard key={note.id} note={note} />
  //   </Grid>
  // ));

  const renderNoteItems = () => {
    props.itemsList.map(item => console.log('note item: ', item));
    // return props.itemsList.map(item => <li>{item.text}</li>)
  }

  console.log('props in note: ', props.itemsList)

  return (
    // <Grid container justify="flex-start" alignItems="center">
    //   {renderNoteItems}
    // </Grid>
    <div>
      {renderNoteItems}
    </div>
  );
};