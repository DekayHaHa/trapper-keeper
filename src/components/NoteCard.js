import React, {Component} from 'react';
import { connect } from 'react-redux';
import { Card, Tooltip, IconButton, CardContent, CardHeader } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import { Link } from 'react-router-dom';
import { NoteItems } from './NoteItems';
import Icon from '@material-ui/core/Icon';
import { deleteNote } from '../thunks/deleteNote';

class NoteCard extends Component {

    render() {
        const { note, classes, deleteNote } = this.props;
        console.log('note card props: ', this.props.note)
      return (
        <Tooltip title='Edit Note' placement='bottom' enterDelay={500}>
            <Card className={classes.card}>
                <Link to='/' className={classes.delete}>
                    <Tooltip title='Delete Note'>
                    <IconButton onClick={() => deleteNote(note.id)}>
                        <Icon>delete</Icon>
                    </IconButton>
                    </Tooltip>
                </Link>
                <Link to={`/api/notes/${note.id}`} className={classes.link}>
                    <CardHeader title={note.title} classes={{ title: classes.title }} />
                    <CardContent>
                        <ul>
                            <NoteItems noteItems={note.itemsList} />
                        </ul>
                    </CardContent>
                  </Link>
            </Card>
        </Tooltip>
      )
    }
}

const styles = {
    link: {
        textDecoration: 'none'
    },
    card: {
        margin: 12
    },
    title: {
        color: '#515151'
    },
    delete: {
        float: 'right',
        margin: '5px'
    }
};

export const mapDispatchToProps = (dispatch) => ({
	deleteNote: (id) => dispatch(deleteNote(id))
});

export default withStyles(styles)(connect(null, mapDispatchToProps)(NoteCard));
