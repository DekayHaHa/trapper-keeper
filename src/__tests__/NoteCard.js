import React from 'react';
import { NoteCard, mapDispatchToProps } from '../components/NoteCard';
import { shallow } from 'enzyme';
import { deleteNote } from '../thunks/deleteNote';

jest.mock('../thunks/deleteNote');

describe('NoteCard', () => {
  let mockItemsList;
  let wrapper;
  let mockFn;
  let mockClasses;
  let mockNote;

  beforeEach(() => {
    mockClasses = { title: 'im an object' }
    mockFn = jest.fn();
    mockNote = { title: 'title 1', itemsList: mockItemsList }
    mockItemsList = [{ text: "item", isComplete: false, id: 1554329952339 }]
    wrapper = shallow(<NoteCard classes={mockClasses} title={'title'} note={mockNote} deleteNote={mockFn} />);
  });

  it('should match snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke deleteNote on removeNote', () => {
    wrapper.find('.icon-btn').simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a deleteNote action', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = deleteNote(2);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.deleteNote(2);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  });
});