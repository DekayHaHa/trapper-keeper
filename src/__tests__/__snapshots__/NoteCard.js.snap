// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteCard should match snapshot with all data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoteCard
    classes={
      Object {
        "title": "im an object",
      }
    }
    deleteNote={[MockFunction]}
    note={
      Object {
        "itemsList": undefined,
        "title": "title 1",
      }
    }
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Card)>
        <Link
          to="/"
        >
          <WithStyles(Tooltip)
            title="Delete Note"
          >
            <WithStyles(IconButton)
              className="icon-btn"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                delete
              </WithStyles(Icon)>
            </WithStyles(IconButton)>
          </WithStyles(Tooltip)>
        </Link>
        <Link
          to="/api/notes/undefined"
        >
          <WithStyles(CardHeader)
            classes={
              Object {
                "title": "im an object",
              }
            }
            title="title 1"
          />
          <WithStyles(CardContent)>
            <ul>
              <WithStyles(NoteItems) />
            </ul>
          </WithStyles(CardContent)>
        </Link>
      </WithStyles(Card)>,
      "enterDelay": 500,
      "id": undefined,
      "placement": "bottom",
      "title": "Edit Note",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Link
            to="/"
          >
            <WithStyles(Tooltip)
              title="Delete Note"
            >
              <WithStyles(IconButton)
                className="icon-btn"
                onClick={[Function]}
              >
                <WithStyles(Icon)>
                  delete
                </WithStyles(Icon)>
              </WithStyles(IconButton)>
            </WithStyles(Tooltip)>
          </Link>,
          <Link
            to="/api/notes/undefined"
          >
            <WithStyles(CardHeader)
              classes={
                Object {
                  "title": "im an object",
                }
              }
              title="title 1"
            />
            <WithStyles(CardContent)>
              <ul>
                <WithStyles(NoteItems) />
              </ul>
            </WithStyles(CardContent)>
          </Link>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Tooltip)
              title="Delete Note"
            >
              <WithStyles(IconButton)
                className="icon-btn"
                onClick={[Function]}
              >
                <WithStyles(Icon)>
                  delete
                </WithStyles(Icon)>
              </WithStyles(IconButton)>
            </WithStyles(Tooltip)>,
            "className": undefined,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(IconButton)
                className="icon-btn"
                onClick={[Function]}
              >
                <WithStyles(Icon)>
                  delete
                </WithStyles(Icon)>
              </WithStyles(IconButton)>,
              "title": "Delete Note",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(Icon)>
                  delete
                </WithStyles(Icon)>,
                "className": "icon-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "delete",
                },
                "ref": null,
                "rendered": "delete",
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(CardHeader)
                classes={
                  Object {
                    "title": "im an object",
                  }
                }
                title="title 1"
              />,
              <WithStyles(CardContent)>
                <ul>
                  <WithStyles(NoteItems) />
                </ul>
              </WithStyles(CardContent)>,
            ],
            "className": undefined,
            "to": "/api/notes/undefined",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "classes": Object {
                  "title": "im an object",
                },
                "title": "title 1",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ul>
                  <WithStyles(NoteItems) />
                </ul>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <WithStyles(NoteItems) />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "noteItems": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "ul",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Card)>
          <Link
            to="/"
          >
            <WithStyles(Tooltip)
              title="Delete Note"
            >
              <WithStyles(IconButton)
                className="icon-btn"
                onClick={[Function]}
              >
                <WithStyles(Icon)>
                  delete
                </WithStyles(Icon)>
              </WithStyles(IconButton)>
            </WithStyles(Tooltip)>
          </Link>
          <Link
            to="/api/notes/undefined"
          >
            <WithStyles(CardHeader)
              classes={
                Object {
                  "title": "im an object",
                }
              }
              title="title 1"
            />
            <WithStyles(CardContent)>
              <ul>
                <WithStyles(NoteItems) />
              </ul>
            </WithStyles(CardContent)>
          </Link>
        </WithStyles(Card)>,
        "enterDelay": 500,
        "id": undefined,
        "placement": "bottom",
        "title": "Edit Note",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Link
              to="/"
            >
              <WithStyles(Tooltip)
                title="Delete Note"
              >
                <WithStyles(IconButton)
                  className="icon-btn"
                  onClick={[Function]}
                >
                  <WithStyles(Icon)>
                    delete
                  </WithStyles(Icon)>
                </WithStyles(IconButton)>
              </WithStyles(Tooltip)>
            </Link>,
            <Link
              to="/api/notes/undefined"
            >
              <WithStyles(CardHeader)
                classes={
                  Object {
                    "title": "im an object",
                  }
                }
                title="title 1"
              />
              <WithStyles(CardContent)>
                <ul>
                  <WithStyles(NoteItems) />
                </ul>
              </WithStyles(CardContent)>
            </Link>,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(Tooltip)
                title="Delete Note"
              >
                <WithStyles(IconButton)
                  className="icon-btn"
                  onClick={[Function]}
                >
                  <WithStyles(Icon)>
                    delete
                  </WithStyles(Icon)>
                </WithStyles(IconButton)>
              </WithStyles(Tooltip)>,
              "className": undefined,
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(IconButton)
                  className="icon-btn"
                  onClick={[Function]}
                >
                  <WithStyles(Icon)>
                    delete
                  </WithStyles(Icon)>
                </WithStyles(IconButton)>,
                "title": "Delete Note",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <WithStyles(Icon)>
                    delete
                  </WithStyles(Icon)>,
                  "className": "icon-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "delete",
                  },
                  "ref": null,
                  "rendered": "delete",
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(CardHeader)
                  classes={
                    Object {
                      "title": "im an object",
                    }
                  }
                  title="title 1"
                />,
                <WithStyles(CardContent)>
                  <ul>
                    <WithStyles(NoteItems) />
                  </ul>
                </WithStyles(CardContent)>,
              ],
              "className": undefined,
              "to": "/api/notes/undefined",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "classes": Object {
                    "title": "im an object",
                  },
                  "title": "title 1",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <ul>
                    <WithStyles(NoteItems) />
                  </ul>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <WithStyles(NoteItems) />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "noteItems": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "ul",
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
